### PR TITLE
FrameDispatcherTest

### PR DESCRIPTION
**Pull request by igordmn**

### CODE REVIEW

File: samples/SkijaInjectSample/build.gradle.kts [hunk 1]
```diff
@@ -64,4 +67,8 @@ tasks.withType<Test> {
         systemProperty("sun.java2d.dpiaware", "false")
         systemProperty("sun.java2d.uiScale", "1")
     }
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    kotlinOptions.freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptI"
```
Comment (olonho) [inline samples/SkijaInjectSample/build.gradle.kts:73]: kotlin.RequiresOptI*n*. Why it is actually needed, just to get rid of warnings?

File: samples/SkijaInjectSample/src/test/kotlin/org/jetbrains/skiko/FrameDispatcherTest.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        lateinit var frameDispatcher: FrameDispatcher
+        frameDispatcher = FrameDispatcher(scope = this) {
+            frameCount++
+            frameDispatcher.scheduleFrame()
+            frameDispatcher.scheduleFrame()
+            frameDispatcher.scheduleFrame()
+        }
+        frameDispatcher.scheduleFrame()
+
+        yield()
```
Comment (olonho) [inline samples/SkijaInjectSample/src/test/kotlin/org/jetbrains/skiko/FrameDispatcherTest.kt:115]: Is this test actually reliable?

File: samples/SkijaInjectSample/build.gradle.kts (hunk 1 above)
Comment (igordmn) [reply samples/SkijaInjectSample/build.gradle.kts:73]: Without it there is a warning:

"This class can only be used with the compiler argument ‘-Xopt-in=kotlin.RequiresOptIn’"

https://kotlinlang.org/docs/reference/opt-in-requirements.html#experimental-status-of-the-opt-in-requirements

File: samples/SkijaInjectSample/src/test/kotlin/org/jetbrains/skiko/FrameDispatcherTest.kt (hunk 2 above)
Comment (igordmn) [reply samples/SkijaInjectSample/src/test/kotlin/org/jetbrains/skiko/FrameDispatcherTest.kt:115]: Yes, we should dispatch frame immediately after any suspend point. This is the requirement for FrameDispatcher.

File: samples/SkijaInjectSample/build.gradle.kts (hunk 1 above)
Comment (olonho) [reply samples/SkijaInjectSample/build.gradle.kts:73]: OK, just put the proper opt-in name then.

Comment (igordmn) [reply samples/SkijaInjectSample/build.gradle.kts]: Oh, thanks. I didn't notice that I lost letter in the commit :)

**Review by olonho — APPROVED** [review]

**Pull request merged**